roles/kubernetes/control-plane/templates/k8s-certs-renew.sh.j2

```diff
@@ -1,7 +1,30 @@
 #!/bin/bash
 
-echo "## Expiration before renewal ##"
-{{ bin_dir }}/kubeadm certs check-expiration
+echo "## Check Expiration before renewal ##"
+
+do_renew=1 # default: always do renewal
+days_buffer=7 # set a time margin, should not renew at the last moment
+calendar={{ auto_renew_certificates_systemd_calendar }}
+next_time=$(systemd-analyze calendar "${calendar}" | grep '^ *Next elapse:' | awk -F': ' '{print $2}')
+# or `systemctl show k8s-certs-renew.timer  -p NextElapseUSecRealtime --value`
+
+if [ "${next_time}" == ""]; then
+	echo "Skip expiry comparison due to fail to parse next elapse from systemd calendar"
+	{{ bin_dir }}/kubeadm certs check-expiration
```

Comment:
That'd mean adding a renew certs here (no needs for check-epxiration)

Reply:
ok